sklearn/linear_model/_base.py

```diff
@@ -518,7 +520,23 @@ def fit(self, X, y, sample_weight=None):
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
-        if sp.issparse(X):
+        if self.solver == "cholesky":
```

Comment:
At this point, X could be a sparse matrix or a dense numpy array, but there are no tests for sparse input with cholesky as yet

Reply:
Updated.